gusto/state.py

```diff
@@ -210,12 +214,74 @@ def setup_dump(self, pickup=False):
+        self.diagnostics_filename = self.dumpdir+"/diagnostics.nc"
+        self.pointdata_filename = self.dumpdir+"/point_data.nc"
+
+        if not pickup:
+            self.setup_diagnostics_output()
+            self.setup_pointdata_output()
+
+    def setup_diagnostics_output(self):
+
+        # setup diagnostics netcdf file
+        diagnostics_data = Dataset(self.diagnostics_filename, "w")
+        # some file info
+        diagnostics_data.description = "Diagnostics data for simulation %s" % self.output.dirname
+        diagnostics_data.history = "Created " + time.ctime()
+        diagnostics_data.source = "Output from Gusto model"
+        # create time dimension - has size None because we will append
+        # to variables along this dimension
+        diagnostics_data.createDimension("time", None)
+        # create time variable so that we can set the values time values
+        times = diagnostics_data.createVariable("time", "f8", ("time",))
+        times.units = "seconds"
+        # create a group for each field - each group will contain the
+        # a variable for each diagnostic
+        for field in self.diagnostics.fields:
+            grp = diagnostics_data.createGroup(field)
+            for diagnostic in self.diagnostics.available_diagnostics:
+                grp.createVariable(diagnostic, 'f8', ('time'))
+        # close the file
+        diagnostics_data.close()
+
+    def setup_pointdata_output(self):
+
+        # setup point data netcdf file
+        point_data = Dataset(self.pointdata_filename, "w")
+        point_data.description = "Point data for simulation %s" % self.output.dirname
+        point_data.history = "Created " + time.ctime()
+        point_data.source = "Output from Gusto model"
+        # create time dimension - has size None because we will append
+        # to variables along this dimension
+        point_data.createDimension("time", None)
+        # create time variable so that we can set the values time values
+        times = point_data.createVariable("time", "f8", ("time",))
+        times.units = "seconds"
+        # create a group for each field - each group will have dimensions
+        # set according to the information in plist
+        for field, points in self.output.point_data:
+            grp = point_data.createGroup(field)
+            # start list of dimensions that point data will have
+            dim_names = ["time"]
+            # get number of points in each direction
+            npts, dim = points.shape
+            grp.createDimension("points", npts)
+            grp.createDimension("geometric_dimension", dim)
+            var = grp.createVariable("points", points.dtype,
+                                     ("points", "geometric_dimension"))
+            var[:] = points
+            # finally, create field variable
+            grp.createVariable(field, self.fields(field).dat.data.dtype,
+                               ("time", "points"))
```

Comment:
Use `self.fields(field).dat.dtype`.  That way you don't access the data (forcing evaluation).